app/services/creators/proceeding_types_creator.rb

```diff
@@ -1,44 +1,34 @@
 module Creators
-  class ProceedingTypesCreator < BaseCreator
-    attr_accessor :proceeding_types
-
-    def initialize(assessment_id:, proceeding_types_params:)
-      super()
-      @assessment_id = assessment_id
-      @proceeding_types_params = proceeding_types_params
-    end
-
-    def call
-      if json_validator.valid?
-        create_records
-      else
-        errors.concat(json_validator.errors)
+  class ProceedingTypesCreator
+    Result = Struct.new :errors, keyword_init: true do
+      def success?
+        errors.empty?
       end
-      self
     end
 
-  private
-
-    attr_reader :assessment_id
+    class << self
+      def call(assessment:, proceeding_types_params:)
+        create_records assessment:, proceeding_types_params:
+      end
 
-    def create_records
-      create_proceeding_types
-    rescue CreationError => e
-      errors << e.errors
-    end
+    private
 
-    def create_proceeding_types
-      self.proceeding_types = assessment.proceeding_types.create!(proceeding_types_attributes)
-    rescue StandardError => e
-      raise CreationError, "#{e.class} - #{e.message}"
-    end
+      def create_records(assessment:, proceeding_types_params:)
+        create_proceeding_types(assessment:, proceeding_types_params:)
+        Result.new(errors: []).freeze
+      # rescue ActiveRecord::RecordInvalid => e
+      #   Result.new(errors: e.record.errors.full_messages).freeze
+      rescue ActiveRecord::RecordNotUnique => e
```

Comment:
@starswan What do you think of rescuing ActiveRecord::RecordInvalid ?

Reply:
I've added a validation so it now raises RecordInvalid